perf_test/sparse/KokkosSparse_block_pcg.cpp

```diff
@@ -0,0 +1,743 @@
+    	block_size = atoi( argv[++i] );
+    }
+
+    else if ( 0 == strcasecmp( argv[i] , "--iteration" ) ) {
+    	cg_iteration_limit = atoi( argv[++i] );
+    }
+    else {
+      cmdline[ CMD_ERROR ] = 1 ;
+      std::cerr << "Unrecognized command line argument #" << i << ": " << argv[i] << std::endl ;
+      std::cerr << "OPTIONS\n\t--threads [numThreads]\n\t--openmp [numThreads]\n\t--cuda\n\t--cuda-dev[DeviceIndex]\n\t--mtx[binary_mtx_file]" << std::endl;
+
+      return 0;
+    }
+  }
+
+  if (mtx_bin_file == NULL){
+    std::cerr << "Provide a mtx binary file" << std::endl ;
+    std::cerr << "OPTIONS\n\t--threads [numThreads]\n\t--openmp [numThreads]\n\t--cuda\n\t--cuda-dev[DeviceIndex]\n\t--mtx[binary_mtx_file]" << std::endl;
+
+    return 0;
+  }
+
+
+
+
+
+
+#if defined( KOKKOS_HAVE_PTHREAD )
+
+    if ( cmdline[ CMD_USE_THREADS ] ) {
+      INDEX_TYPE nv = 0, ne = 0;
+      INDEX_TYPE *xadj, *adj;
+      SCALAR_TYPE *ew;
+
+
+      if ( cmdline[ CMD_USE_NUMA ] && cmdline[ CMD_USE_CORE_PER_NUMA ] ) {
+        Kokkos::Threads::initialize( cmdline[ CMD_USE_THREADS ] ,
+                                     cmdline[ CMD_USE_NUMA ] ,
+                                     cmdline[ CMD_USE_CORE_PER_NUMA ] );
+      }
+      else {
+        Kokkos::Threads::initialize( cmdline[ CMD_USE_THREADS ] );
+      }
+
+      KokkosKernels::Impl::read_matrix<INDEX_TYPE,INDEX_TYPE, SCALAR_TYPE> (&nv, &ne, &xadj, &adj, &ew, mtx_bin_file);
+      Kokkos::Threads::print_configuration(std::cout);
+
+      typedef Kokkos::Threads myExecSpace;
+      typedef typename KokkosSparse::CrsMatrix<SCALAR_TYPE, INDEX_TYPE, myExecSpace, void, SIZE_TYPE > crsMat_t;
+
+      typedef typename crsMat_t::StaticCrsGraphType graph_t;
+      typedef typename graph_t::row_map_type::non_const_type row_map_view_t;
+      typedef typename graph_t::entries_type::non_const_type   cols_view_t;
+      typedef typename crsMat_t::values_type::non_const_type values_view_t;
+
+      row_map_view_t rowmap_view("rowmap_view", nv+1);
+      cols_view_t columns_view("colsmap_view", ne);
+      values_view_t values_view("values_view", ne);
+
+      KokkosKernels::Impl::copy_vector<SCALAR_TYPE * , values_view_t, myExecSpace>(ne, ew, values_view);
+      KokkosKernels::Impl::copy_vector<INDEX_TYPE * , cols_view_t, myExecSpace>(ne, adj, columns_view);
+      KokkosKernels::Impl::copy_vector<INDEX_TYPE * , row_map_view_t, myExecSpace>(nv+1, xadj, rowmap_view);
+
+      graph_t static_graph (columns_view, rowmap_view);
+      crsMat_t crsmat("CrsMatrix", nv, values_view, static_graph);
+      delete [] xadj;
+      delete [] adj;
+      delete [] ew;
+
+      values_view_t kok_x_original = create_x_vector<values_view_t>(((nv /block_size) + 1) * block_size, MAXVAL);
+      for (INDEX_TYPE i = nv; i < ((nv /block_size) + 1) * block_size; ++i){
+    	  kok_x_original(i) = 0;
+      }
+      run_experiment<myExecSpace, crsMat_t>(crsmat, kok_x_original, block_size);
+
+      myExecSpace::finalize();
```

Comment:
I think calling `finalize()` for a specific execution space is to be deprecated in Kokkos, @crtrott.